src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -22,6 +23,8 @@ namespace Microsoft.NET.Sdk.Razor.SourceGenerators
     [Generator]
     public partial class RazorSourceGenerator : ISourceGenerator
     {
+        private static readonly ConcurrentDictionary<Guid, IReadOnlyList<TagHelperDescriptor>> _tagHelperCache = new();
```

Comment:
Can you add a comment that says 

// This is a temporary workaround until the compiler adds support for this.